### DISCUSSION TITLE
Iterating over a range is broken

### DISCUSSION THREAD

**Discussion by Kasparsu** [alpinejs/alpine#3332] [1. Help]
Iterating over a range seems to be broken

[](https://codepen.io/kasparsu/pen/gOKygvy)
example 

https://alpinejs.dev/directives/for#iterating-over-a-range
![image](https://user-images.githubusercontent.com/3220262/206783083-1a077cc2-9203-433f-a8ea-db40f10fd920.png)

**Comment by Kasparsu** [comment]
it seems to start working when I add `x-data="{}"` to the element

**Reply by joshhanley** [reply]
@Kasparsu yeah you need to add `x-data` for Alpine to initialise otherwise the `x-for` won't work.

**Reply by Kasparsu** [reply]
@joshhanley yes there seems to be a note about that in documentation but there seems to be nothing about it in section about iterating over a range... Maybe add some text about this there as from example it seems there would be no need for x-data.

**Reply by joshhanley** [reply]
@Kasparsu yeah Alpine basically always needs `x-data`. But I understand how the example is confusing. The intent was to show as little as possible, so it just focuses on the `x-for` syntax.

**Comment by joshhanley** [comment]
This issue has been converted to a discussion as we are now using discussions to track bugs and ideas.
See this announcement for more details https://github.com/alpinejs/alpine/discussions/1423
